content/articles/simulating-a-simple-epidemic-spreading-process-on-graphs/Simulating a Simple Epidemic Spreaging Process on  40a38af8bf6a4743a874f9bdcc487a61.md

```diff
@@ -0,0 +1,519 @@
+    # update the infected vertices (I -> R)
+    # infect (S -> I) a subset of susceptible neighbors of infected vertices based on some rule
+end
+```
+
+We can compute the state of the vertices (nodes) by looping through `vertices(G)`:
+
+```julia
+function get_nodes_state(G)
+    S = [] # susceptible
+    I = [] # infected
+    R = [] # recovered
+    for v in vertices(G)
+        state = get_prop(G, v, :state)
+        if state=="S"
+            push!(S, v)
+        elseif state=="I"
+            push!(I, v)
+        else
+            push!(R, v)
+        end
+    end
+    return S, I, R
+end
+```
+
+First, `update_nodes!` will compute the susceptible neighbors of the infected nodes.
+
+```julia
+function update_nodes!(G, I)
+		# looping through the infected vertices
+    for v in I
+        potential_infected_neighbors = []
+				# looping through their neighbors
+        for u in neighbors(G, v)
+            if get_prop(G, u, :state)=="S"
+								# a susceptible neighbor goes to potential_infected_neighbors
+                push!(potential_infected_neighbors, u)
+            end
+				end
+		end	
+end
+```
+
+Let’s create two functions to update the vertices state:
+
+```julia
+# (I -> R)
+function set_node_recovered!(G, node)
+    set_props!(G, node, Dict(:state=>"R", :color=>"blue"))
+end
+```
+
+```julia
+# (S -> I)
+function set_node_infectious!(G, node)
+    set_props!(G, node, Dict(:state=>"I", :color=>"red"))
+end
+```
+
+Let’s add `set_node_recovered!(G, v)` in `update_nodes!`: as stated before, the infected nodes in the previous step recovers in the next.
+
+```julia
+function update_nodes!(G, I)
+		# looping through the infected vertices
+    for v in I
+				# updating the infected nodes
+				set_node_recovered!(G, v)
+
+        potential_infected_neighbors = []
+				# looping through their neighbors
+        for u in neighbors(G, v)
+            if get_prop(G, u, :state)=="S"
+								# a susceptible neighbor goes to potential_infected_neighbors
+                push!(potential_infected_neighbors, u)
+            end
+				end
+		end	
+end
+```
+
+Now we need to set a rule that will say how we should choose the neighbors that will be infected in the next step. Let’s keep it simple: each infected vertex will infect `r` random neighbors. If the vertex has less than `r` neighbors, it infects all its neighbors.
+
+To select the `r` random neighbors, let’s use the `sample` function from `StatsBase` package. Import `StatsBase`:
+
+```julia
+using StatsBase
+```
+
+We can get a sample of size `r` using `sample(potential_infected_neighbors, r)`. Let’s create a list to store the vertices that are going to be infected and pass the parameter `r` to `update_nodes!`:
+
+```julia
+function update_nodes!(G, I, r)
+		next_infected_nodes = []
+		
+		# rest of function here...
+end
+```
+
+Now we can add the `potential_infected_neighbors` sample to `next_infected_nodes`.
+
+```julia
```

Comment:
Explain every block of code at the end for better understanding, do this for every instance that's necessary.